src/Sylius/Behat/Context/Setup/ProductContext.php

```diff
@@ -657,6 +650,10 @@ private function createProduct($productName, $price = 0, $date = null)
         $productVariant->setPrice($price);
         $productVariant->setCode($product->getCode());
 
+        if ($this->sharedStorage->has('channel')) {
+            $product->addChannel($this->sharedStorage->get('channel'));
+        }
```

Comment:
Why you moved assigning channels here? It's not a concern of this step. The previous place was correct.